src/lightning_app/runners/cloud.py

```diff
@@ -463,6 +465,30 @@ def _project_has_sufficient_credits(self, project: V1Membership, app: Optional[L
 
         return balance >= 1
 
+    @classmethod
+    def load_app_from_file(cls, filepath: str) -> "LightningApp":
+        """This is meant to use only locally for cloud runtime."""
+        try:
+            app = load_app_from_file(filepath, raise_exception=True)
+        except ModuleNotFoundError as e:
+            # this is very generic exception.
+            logger.info("Could not load the app() locally. Starting the app directly on the cloud.")
+            # we want to format the exception as if no frame was on top.
+            exp, val, tb = sys.exc_info()
+            listing = traceback.format_exception(exp, val, tb)
+            # remove the entry for the first frame
+            del listing[1]
+            from lightning_app.testing.helpers import EmptyFlow
+
+            # Create a mocking app.
+            app = LightningApp(EmptyFlow())
```

Comment:
This is too much of a hack and will cause `flow_cloud_compute` not to be detected correctly.

Let's investigate dependencies patching as discussed earlier @manskx, so no `ModuleNotFoundError` is triggered.

Reply:
Created an issue to do it in a following PR: https://github.com/Lightning-AI/lightning/issues/15428